expat/configure.ac

```diff
@@ -183,33 +183,53 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
+AC_ARG_WITH([getrandom],
+  [AS_HELP_STRING([--with-getrandom] [--without-getrandom],
+  [To enforce/dont use the getrandom function in the system @<:@default=check@:>@])],
+  [],
+  [with_getrandom=check])
 
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AS_IF([test "x$with_getrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+     #include <stdlib.h>  /* for NULL */
+     #include <sys/random.h>
+     int main() {
+       return getrandom(NULL, 0U, 0U);
+     }
+     ])],
+     [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+      AC_MSG_RESULT([yes])],
+      [AC_MSG_RESULT([no])
+       AS_IF([test "x$with_getrandom" = xyes],
+       AC_ERROR(enforced the use of sysgetrandom with --with-sysgetrandom but not detected)
+       )])],
+     [])
+
+AC_ARG_WITH([sysgetrandom],
+ [AS_HELP_STRING([--with-sysgetrandom] [--without-sysgetrandom],
+ [To enforce/dont use the sysgetrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_sysgetrandom=check])
+
+AS_IF([test "x$with_sysgetrandom" != xno],
+  [AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
    AC_LINK_IFELSE([AC_LANG_SOURCE([
       #include <stdlib.h>  /* for NULL */
       #include <unistd.h>  /* for syscall */
       #include <sys/syscall.h>  /* for SYS_getrandom */
       int main() {
         syscall(SYS_getrandom, NULL, 0, 0);
         return 0;
-      }
+   }
    ])],
-   [AC_DEFINE([HAVE_SYSCALL_GETRANDOM], [1], [Define to 1 if you have `syscall' and `SYS_getrandom'.])
+   [AC_DEFINE([HAVE_SYSCALL_GETRANDOM], [1], [Define to 1 if you have 'syscall' and 'SYS_getrandom'.])
```

Comment:
Let's not fix imbalanced string quote — backtick left, straight tick right — here because that's [ancient GNU quoting style](https://www.gnu.org/prep/standards/html_node/Quote-Characters.html) and the rest of generated file `expat_config.` will have it like this because tools generate it like that, see:
```console
# fgrep '`' expat_config.h
/* Define to 1 if you have the `arc4random' function. */
/* Define to 1 if you have the `arc4random_buf' function. */
/* Define to 1 if you have the `getpagesize' function. */
/* Define to 1 if you have the `getrandom' function. */
/* Define to 1 if you have the `bsd' library (-lbsd). */
/* Define to 1 if you have a working `mmap' system call. */
/* Define to 1 if you have `syscall' and `SYS_getrandom'. */
/* Define to include code reading entropy from `/dev/urandom'. */
/* Define to empty if `const' does not conform to ANSI C. */
/* Define to `long int' if <sys/types.h> does not define. */
/* Define to `unsigned int' if <sys/types.h> does not define. */
```
So let's keep it consistent to the rest while tools still do it this way.

Reply:
Taken care.